plugins/UM3NetworkPrinting/resources/qml/PrintJobInfoBlock.qml

```diff
@@ -2,13 +2,119 @@ import QtQuick 2.2
 import QtGraphicalEffects 1.0
+import QtQuick.Layouts 1.1
+import QtQuick.Dialogs 1.1
 
 import UM 1.3 as UM
 
 
 Item
 {
     id: base
+
+    function haveAlert() {
+        return printJob.configurationChanges.length !== 0;
+    }
+
+    function alertHeight() {
+        return haveAlert() ? 230 : 0;
+    }
+
+    function alertText() {
+        if (printJob.configurationChanges.length === 0) {
+            return "";
+        }
+
+        var topLine;
+        if (materialsAreKnown(printJob)) {
+            topLine = catalog.i18nc("@label", "The assigned printer, %1, requires the following configuration change(s):").arg(printJob.assignedPrinter.name);
+        } else {
+            topLine = catalog.i18nc("@label", "The printer %1 is assigned, but the job contains an unknown material configuration.").arg(printJob.assignedPrinter.name);
+        }
+        var result = "<p>" + topLine +"</p>";
+
+        for (var i=0; i<printJob.configurationChanges.length; i++) {
+            var change = printJob.configurationChanges[i];
+            var text = "";
+            if (change.typeOfChange === 'material_change') {
+                text = catalog.i18nc("@label", "Change material %1 from %2 to %3.").arg(change.index + 1).arg(change.originName).arg(change.targetName);
+            } else if (change.typeOfChange === 'material_insert') {
+                text = catalog.i18nc("@label", "Load %3 as material %1 (This cannot be overridden).").arg(change.index + 1).arg(change.targetName);
+            } else if (change.typeOfChange === 'print_core_change') {
+                text = catalog.i18nc("@label", "Change print core %1 from %2 to %3.").arg(change.index + 1).arg(change.originName).arg(change.targetName);
+            } else if (change.typeOfChange === 'buildplate_change') {
+                var target_name = formatBuildPlateType(change.target_name);
+                text = catalog.i18nc("@label", "Change build plate to %1 (This cannot be overridden).").arg(target_name);
+            }
+            result += "<p><b>" + text + "</b></p>"; 
+        }
+        return result;
+    }
+
+    function materialsAreKnown(printJob) {
+        var conf0 = printJob.configuration[0];
+        if (conf0 && !conf0.material.material) {
+            return false;
+        }
+
+        var conf1 = printJob.configuration[1];
+        if (conf1 && !conf1.material.material) {
+            return false;
+        }
+
+        return true;
+    }
+
+    function formatBuildPlateType(buildPlateType) {
+        var translationText = "";
+
+        switch (buildPlateType) {
+            case 'glass':
+                translationText = catalog.i18nc("@label", "Glass");
+                break;
+            case 'aluminum':
+                translationText = catalog.i18nc("@label", "Aluminum");
+                break;
+            default:
+                translationText = null;
+        }
+        return translationText;
+    }
+
+    function formatPrintJobName(name) {
```

Comment:
This shouldn't be in QML in the first place (This should be in the model, where the printJob object is created). This can also be done dynamicly without hardcoding extentions.
It's also a legacy thing, so document it (newer printers don't show this)

Reply:
This is already being fixed in my branch from this branch where I'm rewriting the Print Info Block from scratch to be much cleaner. Don't spend more time reviewing this for now.